Core/Core/Helpers/Http.cs

```diff
@@ -128,23 +131,24 @@ public static async Task<bool> Ping(string hostnameOrAddress)
   }
 
   /// <summary>
-  /// Pings and tries gettign data from a specific address to verify it's online. Retries 3 times.
+  ///
   /// </summary>
-  /// <param name="address">The address to use</param>
-  /// <returns>True if the the status code is successful, false otherwise.</returns>
-  public static async Task<bool> HttpPing(string address)
+  /// <param name="uri">The URI that should be pinged</param>
+  /// <exception cref="HttpRequestException">Failed to ping <paramref name="uri"/></exception>
+  internal static async Task<HttpResponseMessage> HttpPing(Uri uri)
   {
-    SpeckleLog.Logger.Information("HttpPinging {address}", address);
     try
     {
       using var httpClient = GetHttpProxyClient();
-      var response = await httpClient.GetAsync(address).ConfigureAwait(false);
-      return response.IsSuccessStatusCode;
+      HttpResponseMessage response = await httpClient.GetAsync(uri).ConfigureAwait(false);
+      response.EnsureSuccessStatusCode();
+      SpeckleLog.Logger.Warning("Successfully pinged {uri}", uri);
+      return response;
     }
-    catch (Exception ex)
+    catch (HttpRequestException ex)
     {
-      SpeckleLog.Logger.Warning(ex, "Exception while pinging: {message}", ex.Message);
-      return false;
+      SpeckleLog.Logger.Warning(ex, "Ping to {uri} was unsuccessful: {message}", uri, ex.Message);
+      throw new HttpRequestException($"Ping to {uri} was unsuccessful", ex);
```

Comment:
Do we need to weap the HttpRequestException with this to provide the context?

Reply:
Since this is method effectively a helper wrapper around `httpClient.GetAsync`, I wanted to keep it to using the `HttpResponseMessage` return OR  `HttpRequestException` throw as this fits with other HttpClient functions.

This method is internal, so we aren't too concerned with public API. But easy enough change if you'd prefer me to use a different type.

I think it could be quite dangerous to start using `SpeckleException` everywhere inside our private/internal implementation (e.g. if a `SpeckleException` thrown from private/internal implementation bubbles up too high by mistake (to public callers) we'd never really know, and this could hide bugs. Using lower level types might allow us to catch this category of mistake)